### PR TITLE
8304922: [testbug] SliderTooltipNPETest fails on Linux

### PR DESCRIPTION
**Pull request by karthikpandelu**
The test was failing on the first run and then it was not failing. Cause for the failure looked to be the UI state not getting updated. Hence added call to `Toolkit` `firePulse` method.

Able to run the test consistently without failure in Linux after the fix.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8304922](https://bugs.openjdk.org/browse/JDK-8304922): [testbug] SliderTooltipNPETest fails on Linux (**Bug** - P4)

### Reviewers
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1119/head:pull/1119` \
`$ git checkout pull/1119`

Update a local copy of the PR: \
`$ git checkout pull/1119` \
`$ git pull https://git.openjdk.org/jfx.git pull/1119/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1119`

View PR using the GUI difftool: \
`$ git pr show -t 1119`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1119.diff">https://git.openjdk.org/jfx/pull/1119.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1119#issuecomment-1527477502)

### CODE REVIEW

**Review by kevinrushforth — COMMENTED** [review]
It doesn't work for me.

File: tests/system/src/test/java/test/robot/javafx/scene/SliderTooltipNPETest.java [hunk 1]
```diff
@@ -94,6 +94,7 @@ private void dragSliderAfterTooltipDisplayed(int dragDistance) throws Exception
                                 slider.getLayoutX() + slider.getLayoutBounds().getWidth()/2),
                             (int)(scene.getWindow().getY() + scene.getY() +
                                 slider.getLayoutY() + slider.getLayoutBounds().getHeight()/2));
+            Toolkit.getToolkit().firePulse();
```
[the remark this answers is not in the document]
Comment (karthikpandelu) [reply tests/system/src/test/java/test/robot/javafx/scene/SliderTooltipNPETest.java:97]: Oh Okay! Let me check on that again.
Looks like the failure is not consistent in all the Linux systems. Even without the fix it was failing only on the first run for me in VM as well as in Ubuntu workstation.

**Comment by karthikpandelu** [comment]
Since `consumeAutoHidingEvents` property value is asserted in the test, I'm suspecting that it is getting asserted before the mouse moves away from the slider and `consumeAutoHidingEvents` property value is reset to original value.
Since the test is not failing consistently for me in Ubuntu, I believe possibility of product issue is less. Hence  added a latch and small delay to make sure that the property value is asserted only after the mouse is moved.
@kevinrushforth  could you please check this fix.
I have ran the test in Ubuntu 22.04 VM multiple times and test didn't fail.

**Comment by karthikpandelu** [comment]
@kevinrushforth could you please check the updated fix whenever possible?

**Review by kevinrushforth — CHANGES_REQUESTED** [review]
It still fails the first time I run it. I ran it on a physical machine, and I now know why. I can make it fail every time by moving the mouse away from the center of the screen (so that it is outside of where the window will appear).

The problem is that the tooltip will not show up on Linux if you directly move the cursor with Robot to be over the control without it ever having been somewhere else in the window first. So the only reason it works on subsequent runs is that the mouse is left in a position such that it will be over the window when the test is run the next time.

In general, it's also a good idea to click in the window to ensure that it is activated. The only other  robot-based test that shows a Tooltip first moves the mouse to the upper-left corner of the scene and then does a mouse click. Adding that code to your patch causes it to work.

So I recommend adding the following to the test (in addition to the changes you already made, which look good).

```
diff --git a/tests/system/src/test/java/test/robot/javafx/scene/SliderTooltipNPETest.java b/tests/system/src/test/java/test/robot/javafx/scene/SliderTooltipNPETest.java
--- a/tests/system/src/test/java/test/robot/javafx/scene/SliderTooltipNPETest.java
+++ b/tests/system/src/test/java/test/robot/javafx/scene/SliderTooltipNPETest.java
@@ -88,6 +88,13 @@ public class SliderTooltipNPETest {
     }
 
     private void dragSliderAfterTooltipDisplayed(int dragDistance) throws Exception {
+        Util.runAndWait(() -> {
+            // Click somewhere in the Stage to ensure that it is active
+            robot.mouseMove((int)(scene.getWindow().getX() + scene.getX()),
+                            (int)(scene.getWindow().getY() + scene.getY()));
+            robot.mouseClick(MouseButton.PRIMARY);
+        });
+
         Thread.sleep(1000); // Wait for slider to layout
 
         Util.runAndWait(() -> {
```

**Comment by karthikpandelu** [comment]
Thanks for the suggestion @kevinrushforth. Made changes according to your comments. Please check.

**Review by kevinrushforth — APPROVED** [review]

**Pull request closed without merging**